java/android/build-nnstreamer-android-lib.sh

```diff
@@ -0,0 +1,618 @@
+fi
+
+if [[ $enable_decoder_flatbuf == "yes" ]]; then
+    echo "Build with flatbuffers v$decoder_flatbuf_ver for the decoder sub-plugin"
+fi
+
+# Set library name
+nnstreamer_lib_name="nnstreamer"
+
+if [[ $build_type != "all" ]]; then
+    nnstreamer_lib_name="$nnstreamer_lib_name-$build_type"
+fi
+
+echo "NNStreamer library name: $nnstreamer_lib_name"
+
+# Android SDK (Set your own path)
+if [[ -z "$android_sdk_dir" ]]; then
+    [ -z "$ANDROID_SDK_ROOT" ] && echo "Need to set ANDROID_SDK_ROOT." && exit 1
+    android_sdk_dir=$ANDROID_SDK_ROOT
+fi
+
+if [[ -z "$android_ndk_dir" ]]; then
+    [ -z "$ANDROID_NDK_ROOT" ] && echo "Need to set ANDROID_NDK_ROOT." && exit 1
+    android_ndk_dir=$ANDROID_NDK_ROOT
+fi
+
+echo "Android SDK: $android_sdk_dir"
+echo "Android NDK: $android_ndk_dir"
+
+echo "Patching NDK source"
+# See: https://github.com/nnstreamer/nnstreamer/issues/2899
+
+sed -z -i "s|struct AMediaCodecOnAsyncNotifyCallback {\n      AMediaCodecOnAsyncInputAvailable  onAsyncInputAvailable;\n      AMediaCodecOnAsyncOutputAvailable onAsyncOutputAvailable;\n      AMediaCodecOnAsyncFormatChanged   onAsyncFormatChanged;\n      AMediaCodecOnAsyncError           onAsyncError;\n};|typedef struct AMediaCodecOnAsyncNotifyCallback {\n      AMediaCodecOnAsyncInputAvailable  onAsyncInputAvailable;\n      AMediaCodecOnAsyncOutputAvailable onAsyncOutputAvailable;\n      AMediaCodecOnAsyncFormatChanged   onAsyncFormatChanged;\n      AMediaCodecOnAsyncError           onAsyncError;\n} AMediaCodecOnAsyncNotifyCallback;|" $android_ndk_dir/toolchains/llvm/prebuilt/*/sysroot/usr/include/media/NdkMediaCodec.h
+
+# GStreamer prebuilt libraries for Android
+# Download from https://gstreamer.freedesktop.org/data/pkg/android/
+if [[ -z "$gstreamer_dir" ]]; then
+    [ -z "$GSTREAMER_ROOT_ANDROID" ] && echo "Need to set GSTREAMER_ROOT_ANDROID." && exit 1
+    gstreamer_dir=$GSTREAMER_ROOT_ANDROID
+fi
+
+echo "GStreamer binaries: $gstreamer_dir"
+
+# NNStreamer root directory
+if [[ -z "$nnstreamer_dir" ]]; then
+    [ -z "$NNSTREAMER_ROOT" ] && echo "Need to set NNSTREAMER_ROOT." && exit 1
+    nnstreamer_dir=$NNSTREAMER_ROOT
+fi
+
+echo "NNStreamer root directory: $nnstreamer_dir"
+
+# ML API root directory
+if [[ -z "$ml_api_dir" ]]; then
+    [ -z "$ML_API_ROOT" ] && echo "Need to set ML_API_ROOT." && exit 1
+    ml_api_dir=$ML_API_ROOT
+fi
+
+echo "ML API root directory: $ml_api_dir"
+
+# Build result directory
+if [[ -z "$result_dir" ]]; then
+    result_dir=$ml_api_dir/android_lib
+fi
+
+mkdir -p $result_dir
+
+echo "Start to build NNStreamer library for Android (build $build_type)"
+pushd $ml_api_dir
+
+# Make directory to build NNStreamer library
+build_dir=build_android_$build_type
+mkdir -p $build_dir
+
+
+# Change project name from api to nnnstreamer
+svn --force export https://github.com/nnstreamer/nnstreamer-android-resource/trunk/android_api ./$build_dir
+mv ./$build_dir/api/ ./$build_dir/nnstreamer/
```

Comment:
IMO, this is unnecessary if new PR in android-res repo is merged.

1. taos-ci repo : remove api build in nnstreamer and new script for ml-api repo
2. nnstreamer-android-res repo : change directory (api > nnstreamer)
3. AWS : enable ci/android build for ml-api repo
4. ml-api repo : check this PR can successfully build android library
5. nnstreamer repo : remove api directory